pkg/kubernetes/kubernetes.go

```diff
@@ -323,15 +323,27 @@ func createCustomResourceDefinition(ctx context.Context, crd *apiextensionsv1bet
 		log.Errorf("failed to create Kubernetes client using config. Err: %+v", err)
 		return err
 	}
-	_, err = apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Create(ctx, crd, metav1.CreateOptions{})
-	if err == nil {
-		log.Infof("%q CRD created successfully", crd.Name)
-	} else if apierrors.IsAlreadyExists(err) {
-		log.Infof("%q CRD already exists", crd.Name)
-		return nil
+
+	crdName := newCrd.ObjectMeta.Name
+	crd, err := apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Get(ctx, crdName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err = apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Create(ctx, newCrd, metav1.CreateOptions{})
+		if err != nil {
+			log.Errorf("Failed to create %q CRD with err: %+v", crdName, err)
+			return err
+		}
+		log.Infof("%q CRD created successfully", crdName)
 	} else {
-		log.Errorf("failed to create %q CRD with err: %+v", crd.Name, err)
-		return err
+		// Update the existing CRD with new CRD
+		crd.Spec = newCrd.Spec
+		crd.Status = newCrd.Status
+		_, err = apiextensionsClientSet.ApiextensionsV1beta1().CustomResourceDefinitions().Update(ctx, crd, metav1.UpdateOptions{})
```

Comment:
Question: Why dont we just update the CRD from the manifest directly?

Reply:
No, we cannot do like that. The manifest spec doesn't have any resource version etc for it to be updated with directly.
We get below error if we do like that

```
Error initializing Cns Operator. Error: customresourcedefinitions.apiextensions.k8s.io "cnsnodevmattachments.cns.vmware.com" is invalid: metadata.resourceVersion: Invalid value: 0x0: must be specified for an update
```

So, we update the spec and status directly on existing crd.

Reply:
So, maybe we should look at how `kubectl apply -f <file_name>` works. I think this is fine too but it is not equivalent to the kubectl command that updates everything.

Reply:
May be i will take it up as a follow up. I will leave it as is for now.

Reply:
Can you file an issue for this so that we can track this enhancement?

Reply:
Why is this an issue? Not sure if the current code has any problem.

Reply:
As long as we are updating only the spec and status in the yaml then thats fine. Once we start updating some other new fields in this CRD, this code would not work and need to be revisited. I hope we do not revisit this.